src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,250 @@
+import { fork } from "child_process";
+import createInstanceName from "./process-name";
+import envPaths from "env-paths";
+import psList from "@trufflesuite/ps-list";
+import { readFile, rm, mkdir, readdir, writeFile } from "fs/promises";
+import path from "path";
+import { FlavorName } from "@ganache/flavors";
+
+export type DetachedInstance = {
+  instanceName: string;
+  pid: number;
+  startTime: number;
+  host: string;
+  port: number;
+  flavor: FlavorName;
+  cmd: string;
+  version: string;
+};
+
+const READY_MESSAGE = "ready";
+
+const START_ERROR = "An error ocurred spawning a detached instance of Ganache:";
+
+const dataPath = new Promise<string>(async (resolve, reject) => {
+  const path = envPaths(`Ganache/instances`).data;
```

Comment:
@tenthirtyone Does this path play nicely with version-check?

Reply:
I stole the path suffix from [persistent-cache.ts](https://github.com/trufflesuite/ganache/blob/develop/src/chains/ethereum/ethereum/src/forking/persistent-cache/persistent-cache.ts#L106)

Reply:
Ah, good call! @tenthirtyone, we might want to use capital `G` `ganache` in version-check :thinking:

Reply:
Ah here I thought you were referring to my comments. 

So I was all "Ah I'm tagged in chat for this PR I approved. I'll check it to finish on Friday" and here I find I am holding up stuff.

Reply:
@davidmurdoch @jeffsmale90 -> My late to the party post mortem here.

I have to make some assumptions about what plays nicely means.

If it means "This is separate from VC entirely, they will never clobber each other" - maybe, should be, probably? `Conf` has a default name for its file but we can also control that name.

If it means "We are going to combine these to reduce the number of disk reads" then...

w/r/t version-check we made the decision to use `Conf` because Truffle used conf and we wanted to support both using the same config at some point, I believe. 

And version-check was never to be tightly coupled to Ganache because it was originally intended to be a separate lib altogether at some point. 

I have no preference either way but need the clarity on whether VC is still intended to standalone or not. 

If it does change I would still like to review version-check as is since it satisfies the original spec and treat the potential new config work as an issue for future work instead of moving the goal post and prolonging the existing review. 

We also talk about VC as being something that is not changed much after releasing it. 

And my two cents -> Our config system should support file-based configs and this should all go in there.